kml.cc

```diff
@@ -392,6 +415,35 @@ void trk_coord(xg_string args, const QXmlStreamAttributes*)
     track_add_wpt(trk_head, trkpt);
     iargs = iargs.mid(consumed);
   }
+
+  /* The track coordinates do not have a time associated with them. This is specified by using:
+   *
+   * <TimeSpan>
+   *   <begin>2017-08-21T17:00:05Z</begin>
+   *   <end>2017-08-21T17:22:32Z</end>
+   * </TimeSpan>
+   *
+   * If this is specified, then SetCreationDate
+   */
+  if ( wpt_timespan_begin.isValid() && wpt_timespan_end.isValid() ) {
+
+	  // Count the number of Waypoints
+	  n = 0;
+	  QUEUE_FOR_EACH(&trk_head->waypoint_list, curr_elem, tmp_elem) {
+		  n++;
+	  }
+
+	  // If there are some Waypoints, then distribute the TimeSpan to all Waypoints
+	  if ( n > 0 ) {
+		  qint64 timespan_ms = wpt_timespan_end.toMSecsSinceEpoch() - wpt_timespan_begin.toMSecsSinceEpoch();
```

Comment:
There isn't an operator-, oddly, but the msecsTo method may play nicer with large timespans.
http://doc.qt.io/qt-5/qdatetime.html#msecsTo